crm_rma_advance_warranty/model/crm_claim_rma.py

```diff
@@ -0,0 +1,134 @@
+                           DEFAULT_SERVER_DATETIME_FORMAT)
+
+
+class claim_line(models.Model):
+
+    _inherit = 'claim.line'
+
+    # Method to calculate warranty limit
+    @api.one
+    @api.model
+    def set_warranty_limit(self):
+        if not self.date_invoice:
+            raise except_orm(
+                _('Error'),
+                _('Cannot find any date for invoice. '
+                  'Must be a validated invoice.'))
+        warning = 'not_define'
+        date_inv_at_server = datetime.strptime(self.date_invoice,
+                                               DEFAULT_SERVER_DATE_FORMAT)
+        if self.warranty_type == 'supplier':
+            if self.prodlot_id:
+                supplier_rec = self.prodlot_id.supplier_id
+                supplier = False
+                for seller_rec in self.product_id.seller_ids:
+                    if supplier_rec.id == seller_rec.name.id:
+                        supplier = supplier_rec
+                        break
+
+                if not supplier:
+                    raise except_orm(
+                        _('Error'),
+                        _('The product has no supplier'
+                          ' configured for %s.' % supplier_rec.name))
+
+                psi_obj = self.env['product.supplierinfo']
+                domain = [('name', '=', supplier.id),
+                          ('product_tmpl_id', '=',
+                           self.product_id.product_tmpl_id.id)]
+                seller = psi_obj.search(domain)
+
+                warranty_duration = seller.warranty_duration
+        else:
+            warranty_duration = self.product_id.warranty
+        limit = self.warranty_limit(date_inv_at_server, warranty_duration)
+        # If waranty period was defined
+        if warranty_duration > 0:
+            claim_date = datetime.strptime(self.claim_id.date,
+                                           DEFAULT_SERVER_DATETIME_FORMAT)
+            if limit < claim_date:
+                warning = 'expired'
+            else:
+                warning = 'valid'
+        self.write(
+            {'guarantee_limit': limit.strftime(DEFAULT_SERVER_DATE_FORMAT),
+             'warning': warning},)
+
+    # Method to calculate warranty return address
+    @api.one
+    @api.model
+    def set_warranty_return_address(self):
+        """Return the partner to be used as return destination and
+        the destination stock location of the line in case of Return.
+
+        We can have various case here:
+            - company or other: return to company partner or
+              crm_return_address_id if specified
+            - supplier: return to the supplier address
+
+        """
+        return_address = None
+        psi_obj = self.env['product.supplierinfo']
+        supplier = False
+        if self.prodlot_id:
+            supplier_rec = self.prodlot_id.supplier_id
+            supplier = False
+            for seller_rec in self.product_id.seller_ids:
```

Comment:
i've taken @LeartS's comments in both methods as mentioned [below](https://github.com/OCA/rma/pull/39#discussion_r36534051)